src/tt.cpp

```diff
@@ -113,32 +285,48 @@ void TranspositionTable::clear() {
 /// to be replaced later. The replace value of an entry is calculated as its depth
 /// minus 8 times its relative age. TTEntry t1 is considered more valuable than
 /// TTEntry t2 if its replace value is greater than that of t2.
+bool TranspositionTable::probe(const Key fullKey, TTEntry &entry) const {
 
-TTEntry* TranspositionTable::probe(const Key key, bool& found) const {
-
-  TTEntry* const tte = first_entry(key);
-  const uint16_t key16 = (uint16_t)key;  // Use the low 16 bits as key inside the cluster
+  const size_t clusterIndex = mul_hi64(fullKey, clusterCount);
+  Cluster& cluster = TT.table[clusterIndex];
+  TTEntryPacked* const tte = &cluster.entry[0];
+  const uint32_t key = entryKeyFromZobrist(fullKey);
+  const uint8_t extraKey = extraKeyFromZobrist(fullKey);
 
   for (int i = 0; i < ClusterSize; ++i)
-      if (!tte[i].key16 || tte[i].key16 == key16)
+  {
+      const uint32_t tteKey = getClusterEntryKey(cluster.keys, i);
+      const uint32_t tteExtraKey = extractBitField<TTEntryPacked::ExtraHash3>(tte[i].bits);
+      const bool occupied = (extractBitField<TTEntryPacked::Move13>(tte[i].bits) != 0);
```

Comment:
this could use a function to extract raw bits of a given bitfield, i.e. without shifting, just apply mask.